src/main/java/kitchenpos/domain/Menu.java

```diff
@@ -1,52 +1,77 @@
 package kitchenpos.domain;
 
 import java.math.BigDecimal;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
 
 public class Menu {
-    private Long id;
-    private String name;
-    private BigDecimal price;
-    private Long menuGroupId;
-    private List<MenuProduct> menuProducts;
 
-    public Long getId() {
-        return id;
+    private final Long id;
+    private final String name;
+    private final BigDecimal price;
+    private final Long menuGroupId;
+    private final List<MenuProduct> menuProducts;
+
+    public Menu(final String name, final BigDecimal price, final Long menuGroupId,
+                final List<MenuProduct> menuProducts) {
+        this(null, name, price, menuGroupId, menuProducts);
+    }
+
+    public Menu(final Long id, final String name, final BigDecimal price, final Long menuGroupId) {
+        this(id, name, price, menuGroupId, new ArrayList<>());
     }
 
-    public void setId(final Long id) {
+    public Menu(final Long id, final String name, final BigDecimal price, final Long menuGroupId,
+                final List<MenuProduct> menuProducts) {
         this.id = id;
+        this.name = name;
+        this.price = price;
+        this.menuGroupId = menuGroupId;
+        this.menuProducts = menuProducts;
     }
 
-    public String getName() {
-        return name;
+    public BigDecimal calculateTotalSum(final Map<Long, Product> products) {
+        BigDecimal totalSum = BigDecimal.ZERO;
+        for (final MenuProduct menuProduct : menuProducts) {
+            final Product product = products.get(menuProduct.getProductId());
+            final BigDecimal multiplePrice = product.multiplePrice(menuProduct.getQuantity());
+            totalSum = totalSum.add(multiplePrice);
+        }
+        return totalSum;
```

Comment:
```suggestion
        return menuProducts.stream()
            .map(it -> products.get(it.getProductId())
                .multiplePrice(it.getQuantity()))
            .reduce(BigDecimal.ZERO, BigDecimal::add);
```

더 좋은 방법은 아닐 수 있으니 적용은 안하셔도 됩니다!)
이런 식으로 stream을 이용할 수도 있다고 하더라구요!